airflow/models/variable.py

```diff
@@ -167,6 +166,35 @@ def set(
 
         This operation overwrites an existing variable.
 
+        :param key: Variable Key
+        :param value: Value to set for the Variable
+        :param description: Description of the Variable
+        :param serialize_json: Serialize the value to a JSON string
+        :param session: Session
+        """
+        Variable._set(
+            key=key, value=value, description=description, serialize_json=serialize_json, session=session
+        )
+        # invalidate key in cache for faster propagation
+        # we cannot save the value set because it's possible that it's shadowed by a custom backend
+        # (see call to check_for_write_conflict above)
+        SecretCache.invalidate_variable(key)
```

Comment:
Should we do it only if DB isolation mode is on? It seems we dont need to do that if there is direct access to DB

Reply:
We also need to do it for another SecretCache

Reply:
What Vincent means is that is no isolation mode is on, the internal API call is made directly in python and effectively the 3 write operations each two times flush the cache. (before and behind the internal API facade.

I could wrap an `if` around if internal API is active and save the effort of a second flush of cache. That would save a few CPU cycles.I would assume a second flush is not tooo expensive compared to the mental load of an additional if statement. I'd leave it to your opinion.... an if around the second flush only make it if internal API active or two flushes but simpler code?

Reply:
I dont much about the cost of `SecretCache.invalidate_variable(key)`. If you tell me the cost is not high then we can leave it as is. But if the cost is considerable (CPU/memory/latency) then yes we should do something about it. Since it seems it is negligible, I am fine keeping it as is

Reply:
It's not expensive at all. It's just pop from a dict.

Reply:
Yeah, the 2 LoC behind the function just are:
```
        if cls._cache is not None:
            cls._cache.pop(f"{cls._VARIABLE_PREFIX}{key}", None)
```